docs/sources/developers/plugins/build-a-streaming-data-source.md

```diff
@@ -0,0 +1,132 @@
+
+By enabling _streaming_ for your data source plugin, you can update your dashboard as soon as new data becomes available.
+
+For example, a streaming data source plugin can connect to a websocket, or subscribe to a message bus, and update the visualization whenever a new message is available.
+
+Let's see how you can add streaming to an existing data source!
+
+Grafana uses [RxJS](https://rxjs.dev/) to continuously send data from a data source to a panel visualization. There's a lot more to RxJS than what's covered in this guide. If you want to learn more, check out the [RxJS documentation](https://rxjs.dev/guide/overview).
+
+1. Enable streaming for your data source in the `plugin.json` file.
+
+   ```json
+   {
+     "streaming": true
+   }
+   ```
+
+1. Change the signature of the `query` method to return an `Observable` from the `rxjs` package.
+
+   ```ts
+   import { Observable } from 'rxjs';
+   ```
+
+   ```ts
+   query(options: DataQueryRequest<MyQuery>): Observable<DataQueryResponse> {
+     // ...
+   }
+   ```
+
+   Note that you also need to remove the `async` keyword from the method signature.
+
+1. Create an `Observable` for each query and combine them all using the `merge` function from the `rxjs` package.
+
+   ```ts
+   import { Observable, merge } from 'rxjs';
+   ```
+
+   ```ts
+   const observables = options.targets.map(target => {
+     return new Observable<DataQueryResponse>(subscriber => {
+       // ...
+     });
+   });
+
+   return merge(...observables);
+   ```
+
+1. In the `subscribe` function, create a `CircularDataFrame`.
+
+   ```ts
+   import { CircularDataFrame } from '@grafana/data';
+   ```
+
+   ```ts
+   const frame = new CircularDataFrame({
+     append: 'tail',
+     capacity: 1000,
+   });
+
+   frame.refId = query.refId;
+   frame.addField({ name: 'time', type: FieldType.time });
+   frame.addField({ name: 'value', type: FieldType.number });
+   ```
+
+   Circular data frames have a limited capacity. When a circular data frame reaches its capacity, the oldest data point is removed.
+
+1. Use `subscriber.next()` to send the updated data frame whenever you receive new updates.
+
+   ```ts
+   const intervalId = setInterval(() => {
+     frame.add({ time: Date.now(), value: Math.random() });
+
+     subscriber.next({
+       data: [frame],
+       key: query.refId,
+     });
+   }, 500);
+
+   return () => {
+     clearInterval(intervalId);
+   };
+   ```
+
+   > **Note:** This is a simplified example. In practice, you'd call `subscriber.next` as soon as you receive new data from a websocket or a message bus.
+
+1. Here's the final `query` method.
```

Comment:
Is this a step or the example? I do not see what a user is supposed to do here. If there is no action, then it should not be a numbered step.